setup.py

```diff
@@ -12,9 +12,13 @@
 
 # Package requirements, minimal pinning
 install_requires = ['numpy>=1.15.0', 'networkx', 'click>5', 'plucky>=0.4.3',
-                    'dimod>=0.8.11,<0.11,!=0.10.0,!=0.10.1,!=0.10.2,!=0.10.3,!=0.10.4',
-                    'minorminer>=0.1.7', 'dwave-networkx>=0.6.6', 'dwave-system>=0.7.6',
-                    'dwave-neal>=0.5.1', 'dwave-tabu>=0.2.0', 'dwave-greedy>=0.1.0']
+                    'dimod>=0.9.6,<0.11,!=0.10.0,!=0.10.1,!=0.10.2,!=0.10.3,!=0.10.4',
+                    # Include dwave-preprocessing, as dimod went back and forth with
+                    # including it as a core dependency vs extra. Make it unbounded,
+                    # as it's already constrained by dimod (if required).
+                    'dwave-preprocessing',
```

Comment:
I believe pip's version resolution order will fail if we release a version of `dwave-preprocessing` that's outside of dimod's version constraints. Since it does something like bfs

Reply:
Not sure I understand what you mean?

Reply:
IIRC, it will first install everything listed here using the ranges specified here. So `dwave-preprocessing` will be unbounded. Then it will install the dependencies of the dependencies and discover that dimod (may) require something smaller. I could be misremembering though the order.

Reply:
Perhaps, but dimod doesn't constrain dwave-preprocessing anymore, right?

Reply:
Of course, you're right.

FWIW dimod and dwave-preprocessing are linked at the C++ level, and when we merge https://github.com/dwavesystems/dimod/pull/1043 that will constitute a breaking change for dwave-preprocessing.

Reply:
Good to know. :facepalm: